uber/site_sections/groups.py

```diff
@@ -1,4 +1,101 @@
 from uber.common import *
+from uber.custom_tags import pluralize
+
+
+_entangled_ribbons = set(
+    getattr(c, r, -1) for r in ['BAND', 'DEPT_HEAD_RIBBON', 'PANELIST_RIBBON'])
+
+
+def _is_attendee_disentangled(attendee):
+    """
+    Returns True if the attendee has an unpaid badge and does not have any
+    other roles in the system.
+    """
+    return attendee.paid not in [c.HAS_PAID, c.NEED_NOT_PAY] \
+        and _entangled_ribbons.isdisjoint(attendee.ribbon_ints) \
+        and not attendee.admin_account \
+        and not attendee.shifts
+
+
+def _is_dealer_convertable(attendee):
+    """
+    Returns True if a waitlisted dealer can be converted into a new, unpaid
+    attendee badge.
+    """
+    return attendee.badge_type == c.ATTENDEE_BADGE \
+        and _is_attendee_disentangled(attendee)
+        # It looks like a lot of dealers have helpers that didn't get assigned
```

Comment:
... the first part of this comment is worrying. :\ That should not happen. Good fallback, though.

Reply:
Historically a lot of dealers don't actually fill in their extra badges until **very** late, often after they know if they made the cut.